### PR TITLE
Discord reminder

### PR DESCRIPTION
**Pull request by david30907d**
## Types of changes

- **New feature**
- **Refactoring**

## Description

1. implement discord reminder
2. create `.env` to store credentials
## Checklist:
- [ ] Add test cases to all the changes you introduce
- [x] Run `poetry run pytest` locally to ensure all linter checks pass
- [x] Update the documentation if necessary

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# get the complete env from other volunteers, please
+BIGQUERY_PROJECT=pycontw-225217
+DATATEAM_DISCORD_WEBHOOK=<>
```
Comment (david30907d) [inline .env.template:3]: if you want to implement the discord-related feature, discord me to get the real token , please.

File: README.md [hunk 2]
```diff
@@ -36,14 +36,28 @@ Using Airflow to implement our ETL pipelines
...
+> Please don't use the .env for local development, or it might screw up the production tables.
+
+1. Build docker image:
+    1. First, build a prod image (for prod): `docker build -t pycon_etl:prod --cache-from davidtnfsh/pycon_etl:prod -f Dockerfile .`
+    2. Build dev/test image (for dev/test): `docker build -t pycon_etl:test --cache-from pycon_etl:prod -f Dockerfile.test .`
+2. Fill in some secrets:
+    1. `cp .env.template .env.staging`
+    2. Follow the instruction in `.env.staging` and fill in your secrets
+3. Start the Airflow server:
+    * dev/test: `docker run --rm -p 80:8080 --name airflow  -v $(pwd)/dags:/usr/local/airflow/dags -v $(pwd)/service-account.json:/usr/local/airflow/service-account.json --env-file=./.env pycon_etl:test webserver`
```
Comment (david30907d) [inline README.md:51]: use `--env-file=./.env ` to get all the credentials in container

File: dags/ods/survey_cake/udfs/survey_cake_csv_uploader.py [hunk 3]
```diff
@@ -22,7 +21,7 @@ def __init__(self, filename):
 
     @property
     def bigquery_project(self):
-        return BIGQUERY_PROJECT
+        return os.getenv("BIGQUERY_PROJECT")
```
Comment (david30907d) [inline dags/ods/survey_cake/udfs/survey_cake_csv_uploader.py:24]: get credential from env directly~

File: .env.template (hunk 1 above)
Comment (tai271828) [reply .env.template:3]: please  "contact me"? :)

Comment (tai271828) [reply .env.template:3]: nitpick: "no new line at end of file"

File: Dockerfile.test [hunk 4]
```diff
@@ -1,3 +1,3 @@
-FROM davidtnfsh/pycon_etl:cache
+FROM davidtnfsh/pycon_etl:prod
 ENV AIRFLOW_TEST_MODE=True
 USER root
```
Comment (tai271828) [inline Dockerfile.test:3]: ditto

File: README.md [hunk 5]
```diff
@@ -36,14 +36,28 @@ Using Airflow to implement our ETL pipelines
...
     * service-account.json: Please contact @david30907d using email, telegram or discord.
-3. Setup the Authentication of GCP: <https://googleapis.dev/python/google-api-core/latest/auth.html>
+4. Setup the Authentication of GCP: <https://googleapis.dev/python/google-api-core/latest/auth.html>
     * After invoking `gcloud auth application-default login`, you'll get a credentials.json resides in `/Users/<xxx>/.config/gcloud/application_default_credentials.json`. Invoke `export GOOGLE_APPLICATION_CREDENTIALS="/path/to/keyfile.json"` if you have it.
-4. Give [Toy-Examples](#Toy-Examples) a try
+5. Give [Toy-Examples](#Toy-Examples) a try
 
 ## Deployment
+
+1.
```
Comment (tai271828) [inline README.md:60]: unexpected line?

**Review by tai271828 — APPROVED** [review]
Some nitpicks inline. Please feel free to merge anytime you want.

**Review by tai271828 — CHANGES_REQUESTED** [review]
Wait..README file conflicts.

File: .env.template (hunk 1 above)
Comment (david30907d) [reply .env.template:3]: will add the new line~

File: Dockerfile.test (hunk 4 above)
Comment (david30907d) [reply Dockerfile.test:3]: haha sure. Btw, would u recommend some Dockerfile linter for this kind of nitpicks?

Comment (tai271828) [reply Dockerfile.test]: FYI: most mainstream IDE or editor should support "no end of line" automatic fix. PyCharm, VIM ...etc.

**Pull request merged**